clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/model/EcsServerGroup.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2017 Lookout, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.ecs.model;
+
+import com.netflix.spinnaker.clouddriver.model.Instance;
+import com.netflix.spinnaker.clouddriver.model.ServerGroup;
+
+import java.util.Map;
+import java.util.Set;
+
+public class EcsServerGroup implements ServerGroup {
+
+  String name;
```

Comment:
Do all of these variables need to be public?

Reply:
they are package-private, not public.  This means they can be accessed only from classes that have exactly the same package name

most java people have the "make everything private" philosophy.  Some Java people have the "why do you even care?" philosophy.  Android performance-minded engineers reject getters because they are slower than accessing public fields.  Groovy people have the "make everything public by default" philosophy.  

This really is inconsequential in the grand scheme of things, as this is a model class.  Also, depending on who you ask the guidance regarding best practice would be different.  Bigger fish to fry out there.